linked_list/implementation.py

```diff
@@ -7,34 +8,144 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        self.length = 0
+        
+        if elements is not None and len(elements) != 0:
+            self.start = Node(elements[0])
+            self.length += 1
+            current = self.start
+            for elem in elements[1:]:
+                current.next = Node(elem)
+                current = current.next
+                self.length += 1
+            self.end = current
+        else:
+            self.elements = elements
+            
 
     def __str__(self):
-        pass
+        if not len(self):
+            return '[]'
+        ret_val = '['
+        for elem in self:
+           ret_val += "{}, ".format(elem)
+        #if ret_val[-2:] == ", ":
+        ret_val = ret_val[:-2]
+        ret_val += ']'
+        return ret_val
 
     def __len__(self):
-        pass
-
+        return self.length
+            
     def __iter__(self):
-        pass
+        iter_object = self.start
+        while iter_object:
+            yield iter_object.elem
+            iter_object = iter_object.next
+        raise StopIteration
 
     def __getitem__(self, index):
-        pass
+        if index >= len(self):
+            raise IndexError
+        for idx, elem in enumerate(self):
+            if idx == index:
+                return elem
```

Comment:
Suggestion: since `LinkedList` is already an iterable due to our use of `__iter__`, we can directly get the `elem` at particular `index` by using something like this:
```python
return list(iter(self))[index]
```